packages/story-editor/src/app/story/useStoryReducer/reducers/updatePage.ts

```diff
@@ -31,14 +33,14 @@ import { objectWithout } from './utils';
  * If id doesn't exist, nothing happens.
  *
  * Current page and selection is unchanged.
- *
- * @param {Object} draft Current state
- * @param {Object} payload Action payload
- * @param {number} payload.pageId Page index to update. If null, update current page.
- * @param {number} payload.properties Object with properties to set for given page.
  */
-export const updatePage = (draft, { pageId, properties }) => {
-  const idToUpdate = pageId === null ? draft.current : pageId;
+export const updatePage = (
+  draft: Story,
+  { pageId, properties }: UpdatePageProps
+) => {
+  const idToUpdate = (pageId === null ? draft.current : pageId) as
+    | string
+    | undefined;
```

Comment:
Isn't this inferred?

Reply:
If not, add a todo.

Reply:
For some reason it complains about `any` unless I specifically cast it, even though in IDE it shows the inferred types correctly. Still have to check why it's like this.

Reply:
Hmm , why can `current` be `null`? Is that allowed? Wouldn't most of the editor break down if that ever happened?

Reply:
In this particular case, `current` shouldn't be null at least, so this would make it simpler:

```ts
const idToUpdate = pageId === null ? (draft.current as string) : pageId;